driver/reducecalculation_driver.hpp

```diff
@@ -207,13 +252,26 @@ int ReduceCalculationDriver<Tgpu, Tref>::AddCmdLineArgs()
         "Nan number propagation mode (check the miopenReduceCalculationNanPropagation_t in "
         "miopen.h) (Default=0 to indicate no Nan propagation)",
         "int");
-    inflags.AddInputFlag(
-        "ReduceCalculationOp",
-        'O',
-        "2",
-        "Reduce Calculation Operation Type (check the miopenReduceCalculationOp_t in "
-        "miopen.h) (Default=2 to add the values of the reduced elements)",
-        "int");
+    if(isLogicalCalculation)
+    {
+        inflags.AddInputFlag(
+            "ReduceCalculationOp",
+            'O',
+            "3",
+            "Reduce Calculation Operation Type (check the miopenReduceCalculationOp_t in "
+            "miopen.h) (Default=0 to indicate logical calculation)",
+            "int");
+    }
+    else
+    {
+        inflags.AddInputFlag(
+            "ReduceCalculationOp",
+            'O',
+            "2",
+            "Reduce Calculation Operation Type (check the miopenReduceCalculationOp_t in "
+            "miopen.h) (Default=2 to add the values of the reduced elements)",
+            "int");
+    }
```

Comment:
how user can on-off this isLogicalCalculation mode while you add it to define params?

Reply:
I fixed my solution, `isLogicalCalculation` is determined based on the type of op, which is passed through flag `ReduceCalculationOp`